pkg/controller/hostpathprovisioner/storagepool.go

```diff
@@ -87,7 +87,9 @@ func (r *ReconcileHostPathProvisioner) reconcileStoragePools(logger logr.Logger,
 		sp := r.getStoragePoolForDeployment(cr, &ds)
 		if sp != nil {
 			if _, err := r.createCleanupJobForDeployment(logger, cr, namespace, &ds, sp); err != nil {
-
+				// TODO: should return 'reconcile.Result{}, err' here
+				// But there is currently an issue, tests get 404 on GET node
+				logger.V(3).Info("Error createCleanupJobForDeployment", "pool.Name", sp.Name, "err", err)
```

Comment:
So the code was kind of correct ignoring the error, since there is no reason to create a cleanup job if the node we are trying to clean up is gone. But we should handle it better in `createCleanupJobForDeployment` something like this:
```go
func (r *ReconcileHostPathProvisioner) createCleanupJobForDeployment(logger logr.Logger, cr *hostpathprovisionerv1.HostPathProvisioner, namespace string, deployment *appsv1.Deployment, storagePool *hostpathprovisionerv1.StoragePool) (*corev1.Node, error) {
	node := &corev1.Node{
		ObjectMeta: v1.ObjectMeta{
			Name: deployment.Spec.Template.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms[0].MatchExpressions[0].Values[0],
		},
	}
	if err := r.client.Get(context.TODO(), client.ObjectKeyFromObject(node), node); err != nil {
		if errors.IsNotFound(err) {
			return nil, nil
		}
		return nil, err
	}
	logger.V(3).Info("for node", "name", node.Name)
	if err := r.createCleanupJobForNode(logger, cr, namespace, storagePool, node); err != nil && !errors.IsAlreadyExists(err) {
		return nil, err
	}
	return node, nil
}
```